tests/conftest.py

```diff
@@ -6,27 +6,56 @@
 import pytest
 
 if TYPE_CHECKING:
-    from typing import Any, Callable, Dict, Literal, Optional, Type, Union
+    from typing import Any, Literal, Optional, Protocol, Union
 
     from requests_mock import Mocker
     from utils import ResourceType
 
     from otelib.client import OTEClient
 
-    OTEResponse = Callable[
-        [
-            Union["HTTPMethod", str],
-            str,
-            int,
-            Optional[Union[Dict[str, Any], str]],
-            Optional[dict],
-            Optional[bytes],
-            Optional[Union[dict, str]],
-            Optional[str],
-            Optional[OTEClient],
-        ],
-        None,
-    ]
+    class TestResourceIds(Protocol):
+        """Return a test id for the given `resource_type`."""
+
+        def __call__(self, resource_type: Union[ResourceType, str]) -> str:
+            ...
+
+    class OTEResponse(Protocol):
+        """Use `requests_mock` to mock a response from an OTE services server.
+
+        It will only be ensured that the `endpoint` starts with a forward slash.
+        If it does not, one will be added. Otherwise, `endpoint` is not manipulated.
+        """
+
+        def __call__(
+            self,
+            method: Union["HTTPMethod", str],
+            endpoint: str,
+            status_code: int = 200,
+            params: Optional[Union[dict[str, Any], str]] = None,
+            headers: Optional[dict] = None,
+            return_content: Optional[bytes] = None,
```

Comment:
Names like return_content, return_json, and return_text might be better named response_content, response_json, and response_text respectively. This makes it clearer that these are expected response outputs, not some action the function should take.